gws-core/src/main/java/au/gov/ga/geodesy/port/adapter/rest/SkeletonEndpoint.java

```
@@ -0,0 +1,126 @@
+            // rinexFileHeader.setObserverAgency(observerAgency);
+
+            List<GnssReceiverLogItem> receiverLogItemList =
+                    new ArrayList<GnssReceiverLogItem>(siteLog.getGnssReceivers());
+            receiverLogItemList.sort(Comparator.comparing(GnssReceiverLogItem::getDateInstalled).reversed());
+
+            GnssReceiverLogItem receiverLogItem = receiverLogItemList.get(0);
+            rinexFileHeader.setReceiverSerialNumber(receiverLogItem.getSerialNumber());
+            rinexFileHeader.setReceiverType(receiverLogItem.getType());
+            rinexFileHeader.setReceiverFirmwareVersion(receiverLogItem.getFirmwareVersion());
+
+            List<GnssAntennaLogItem> antennaLogItemList = new ArrayList<GnssAntennaLogItem>(siteLog.getGnssAntennas());
+            antennaLogItemList.sort(Comparator.comparing(GnssAntennaLogItem::getDateInstalled).reversed());
+
+            GnssAntennaLogItem antennaLogItem = antennaLogItemList.get(0);
+            rinexFileHeader.setAntennaSerialNumber(antennaLogItem.getSerialNumber());
+            rinexFileHeader.setAntennaType(antennaLogItem.getType());
+
+            Point position = siteLog.getSiteLocation().getApproximatePosition().getCartesianPosition();
+            rinexFileHeader.setApproxPositionXyz(new MutableTriple<Double, Double, Double>(
+                    position.getCoordinate().x, position.getCoordinate().y, position.getCoordinate().z));
+
+            rinexFileHeader.setAntennaMarkerArp(new MutableTriple<Double, Double, Double>(
+                    antennaLogItem.getMarkerArpUpEcc(),
+                    antennaLogItem.getMarkerArpEastEcc(),
+                    antennaLogItem.getMarkerArpNorthEcc()));
+
+            return ResponseEntity.ok(assembler.toResource(rinexFileHeader));
+        } catch (Exception e) {
+            // In case of invalid site log file, no receiver or no antenna records at all, etc.
+            log.error("Invalid siteLogItem for site " + fourCharId);
```

Doesn't look like the exception gets logged. You should log the exception so we know what was invalid.

---

We can't catch all exceptions and then say "Invalid siteLogItem...", because the exception could be anything. I think we should log the exception without attempting to interpret what went wrong and sent HTTP 500 to the client.